pkg/driver/estargz/estargz.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package estargz
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/containerd/containerd/images/converter"
+	"github.com/containerd/containerd/platforms"
+	"github.com/containerd/stargz-snapshotter/estargz"
+	estargzconvert "github.com/containerd/stargz-snapshotter/nativeconverter/estargz"
+	"github.com/goharbor/acceleration-service/pkg/content"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type Driver struct {
+	cfg map[string]string
+}
+
+func New(cfg map[string]string) (*Driver, error) {
+	return &Driver{cfg}, nil
+}
+
+func (d *Driver) Convert(ctx context.Context, p content.Provider) (*ocispec.Descriptor, error) {
+	opts, docker2oci, allPlatforms, err := getESGZConvertOpts(d.cfg)
+	if err != nil {
+		return nil, err
+	}
+	platformMC := platforms.DefaultStrict()
+	if allPlatforms {
+		platformMC = platforms.All
+	}
```

Comment:
Would it be possible to detect platforms with accessible content blobs, and push the available platforms by default?

Reply:
Because when the server is amd64 and the client is arm64, the user wants the arm64 contents to be converted by default, regardless to the arch of the server.

Reply:
> Would it be possible to detect platforms with accessible content blobs, and push the available platforms by default?

@AkihiroSuda How about always using `platfroms.All`?